libstuff/SDeburr.cpp

```diff
@@ -0,0 +1,137 @@
+#include "SDeburr.h"
+
+#include <bit>
+#include <cstring>
+#include <string>
+
+#include <libstuff/libstuff.h>
+#include <libstuff/sqlite3.h>
+
+using namespace std;
+
+const char* SDeburr::unicodeToAscii(uint32_t codepoint) {
+    if (codepoint >= 0x00C0 && codepoint <= 0x017F) {
+        return SDeburr::UNICODE_TO_ASCII_MAP[codepoint - 0x00C0];
+    }
+
+    if (codepoint == 0x1E9E) {
+        return "SS"; // ẞ
+    }
+
+    return nullptr;
+}
+
+string SDeburr::deburr(const string& input) {
+    const unsigned char* input_bytes = reinterpret_cast<const unsigned char*>(input.c_str());
```

Comment:
Shouldn't need the cast here.

Reply:
yeah, you mentioned this but I get compiler errors if I don't cast here. `input.c_str()` and `input.data()` both return `const char*`, not `const unsigned char*`. The `countl_one` C++ stdlib utility expects a `const unsigned char*`, so I couldn't figure out how to avoid the cast. A bit of research suggested that changing a `char*` to an `unsigned char*` is one of the primary use-cases for `reinterpret_cast` (`static_cast` didn't work).

lmk if you see something I don't here